### PR TITLE
Inactive subscribers improvements [MAILPOET-3774]

### PR DESCRIPTION
**Pull request by lysyjan**
[MAILPOET-3774]

[MAILPOET-3774]: https://mailpoet.atlassian.net/browse/MAILPOET-3774

### CODE REVIEW

File: lib/Subscribers/InactiveSubscribersController.php [hunk 1]
```diff
@@ -168,10 +146,14 @@ private function activateSubscribers(Carbon $thresholdDate, $batchSize) {
         ->findArray();
     } else {
       $idsToActivate = ORM::forTable($subscribersTable)->select("$subscribersTable.id")
-        ->leftOuterJoin($statsOpensTable, "$subscribersTable.id = $statsOpensTable.subscriber_id AND $statsOpensTable.created_at > '$thresholdDate'")
+        ->leftOuterJoin($subscribersTable, "$subscribersTable.id = s2.id AND GREATEST(
+          COALESCE(s2.last_engagement_at, 0),
```
Comment (costasovo) [inline lib/Subscribers/InactiveSubscribersController.php:175]: The same issue with COALESCE parameters as mentioned in the previous comment.

**Review by costasovo — CHANGES_REQUESTED** [review]
Great job! I found an issue with COALESCE parameters, that we should address. Otherwise looks good.
Did you try some performance testing?

File: lib/Subscribers/InactiveSubscribersController.php [hunk 2]
```diff
@@ -124,9 +103,13 @@ private function deactivateSubscribers(Carbon $thresholdDate, $batchSize, $start
     );
 
     $idsToDeactivate = ORM::forTable($inactiveSubscriberIdsTmpTable)->rawQuery("
-      SELECT s.id FROM $inactiveSubscriberIdsTmpTable s
-        LEFT OUTER JOIN $statisticsOpensTable as so ON s.id = so.subscriber_id AND so.created_at > ?
-        WHERE so.id IS NULL",
+      SELECT isi.id FROM $inactiveSubscriberIdsTmpTable isi
+        LEFT OUTER JOIN $subscribersTable as s ON isi.id = s.id AND GREATEST(
+          COALESCE(s.last_engagement_at, 0),
```
[the remark this answers is not in the document]
Comment (lysyjan) [reply lib/Subscribers/InactiveSubscribersController.php:124]: Thanks for the notification. Maybe we should improve our development environment to the strict mode in MySQL too. What do you think?

[the remark this answers is not in the document]
Comment (costasovo) [reply lib/Subscribers/InactiveSubscribersController.php:124]: @lysyjan Good idea. 👍

**Comment by lysyjan** [comment]
> Did you try some performance testing?

Yes, I tried queries on the dump from mailpoet.com, and it works well on my local. I think that the run in batches assures that.

**Review by costasovo — APPROVED** [review]

**Review by veljkho — APPROVED** [review]

**Pull request merged**